contracts/v3/FiatTokenV3.sol

```diff
@@ -0,0 +1,300 @@
+     */
+    function blacklist(address _account) override external onlyBlacklister {
+        _blacklist(_account);
+    }
+
+    /**
+     * @dev Internal function to process additions to the blacklist
+     * @param _account The address to blacklist
+     */
+    function _blacklist(address _account) internal {
+        balances[_account] = balances[_account] | (uint256(1) << 255);
+        emit Blacklisted(_account);
+    }
+
+    /**
+     * @dev Removes account from blacklist
+     * @param _account The address to remove from the blacklist
+     */
+    function unBlacklist(address _account) override external onlyBlacklister {
+       _unBlacklist(_account);
+    }
+
+    /**
+     * @dev Internal function to process removals from the blacklist
+     * @param _account The address to remove from the blacklist
+     */
+    function _unBlacklist(address _account) internal {
+        balances[_account] = balances[_account] & ~(uint256(1) << 255);
+        emit UnBlacklisted(_account);
+    }
+
+    /**
+     * @dev Checks if account is blacklisted
+     * @param _account The address to check
+     */
+    function isBlacklisted(address _account) override external view returns (bool) {
+        return balances[_account] >> 255 == 1;
+    }
+
+    /**
+     * @dev Throws if argument account is blacklisted
+     * @param _account The address to check
+     */
+    modifier notBlacklisted(address _account) override {
+        require(
+            !(balances[_account] >> 255 == 1),
+            "Blacklistable: account is blacklisted"
+        );
+        _;
+    }
+
+    /**
+     * @dev Get token balance of an account
+     * @param account address The account
+     */
+    function balanceOf(address account)
+        external
+        override
+        view
+        returns (uint256)
+    {
+        // return balance without effect of blacklist high bit
+        return balances[account] & ~(uint256(1) << 255);
+    }
+
+    /**
+     * @dev Get token balance of an account
+     * @param account address The account
+     */
+    function _getBalanceAndBlacklistStatus(address account)
+        internal
+        view
+        returns (uint256, bool)
+    {
+        uint256 balance = balances[account];
+        return (balance & ~(uint256(1) << 255), balance >> 255 == 1);
+    }
+
+    /**
+     * @notice Internal function to process transfers
+     * @param from  Payer's address
+     * @param to    Payee's address
+     * @param value Transfer amount
+     */
+    function _transfer(
+        address from,
+        address to,
+        uint256 value
+    ) internal override {
+        require(from != address(0), "ERC20: transfer from the zero address");
+        require(to != address(0), "ERC20: transfer to the zero address");
+
+        (uint256 fromBalance, bool isFromBlacklisted) = _getBalanceAndBlacklistStatus(from);
+
+        require(!isFromBlacklisted, "Blacklistable: account is blacklisted");
```

Comment:
That's right, by moving the blacklist checks here, you can read from the balance storage slot just once, grabbing both the blacklist boolean and the balance.

You'd still get gas savings from keeping all these checks as modifiers, since it would be warm SLOADs instead of cold SLOADs the second time the storage slots are read, but this is more gas efficient.